src/sagemaker/fw_utils.py

```diff
@@ -1045,6 +1043,58 @@ def validate_torch_distributed_distribution(
         raise ValueError(err_msg)
 
 
+def validate_smddp_collectives_support(
+    framework_version, py_version, image_uri, instance_type, instance_count
+):
+    """Check if SMDDP collective backend is supported for current invocation.
+
+    Args:
+        framework_version (str): A string representing the framework version selected.
+        py_version (str): A string representing the python version selected.
+        image_uri (str): A string representing a Docker image URI.
+        instance_type (str): SageMaker instance type.
+        instance_count (int): Number of training instances to use.
+
+    Returns false if:
+        `instance_type` is not in SMDDP_COLLECTIVES_SUPPORTED_INSTANCE_TYPES or
+        `py_version` is not python3 or
+        `framework_version` is not in SMDDP_COLLECTIVES_SUPPORTED_FRAMEWORK_VERSIONS or
+        `instance_count` is not greater than 1
+    """
+    err_msg = ""
+    if not image_uri:
+        # ignore framework_version and py_version if image_uri is set
+        # in case image_uri is not set, then both are mandatory
+        if framework_version not in SMDDP_COLLECTIVES_SUPPORTED_FRAMEWORK_VERSIONS:
+            err_msg += (
+                f"Provided framework_version {framework_version} is not supported."
+                "Please specify one of the supported framework versions:"
+                f" {SMDDP_COLLECTIVES_SUPPORTED_FRAMEWORK_VERSIONS}.\n"
+            )
+        if "py3" not in py_version:
+            err_msg += (
+                f"Provided py_version {py_version} is not supported."
+                "Please specify py_version>=py3.\n"
+            )
+    if instance_type not in SMDDP_COLLECTIVES_SUPPORTED_INSTANCE_TYPES:
+        err_msg += (
+            f"Provided instance_type {instance_type} is not supported."
+            "Please specify one of the supported instance types:"
+            f"{SMDDP_COLLECTIVES_SUPPORTED_INSTANCE_TYPES}.\n"
+        )
+    if instance_count == 1:
+        # Communication backend auto is not supported for single-node jobs
+        err_msg += (
+            "SMDDP Collective backend is not supported for single-node jobs."
+            "Please increase instance_count to be greater than 1.\n"
+        )
+    if not err_msg:
+        return True
+    logger.warning("Could not enable SMDDP Collectives for the training job.\n%s", err_msg)
```

Comment:
To make it more explictit for user, because they set "auto" and they would confused what is this SMDDP message and I don't know what SMDDP is.

How about - Communication backend set to 'auto'. Unable to evaluate AWS infrastructure optimized SMDDP collectives because - err_msg\n Continuing model training with default NCCL communication backend.

something like that, so message gives full context for a user who don't know how 'auto' works.

Reply:
Yeah, that makes more sense. Will update.